### PR TITLE
[13] Preparing training data

### LINKED ISSUE #13: Preparing training data
- Training data (positive samples) provided in the sprint planning deck
- For negative samples, let's initially try sampling randomly from the Crunchbase and GtR datasets (to obtain a balanced sample) + from the other mission classes
- Getters for GtR data
- Getters for training data
- In terms of output format, fine to have a simple table (as opposed to a json)

### PR DESCRIPTION
**Pull request by Jack-0-0**
Closes #13.

This PR adds:
* Loading, processing and making of horizon scout classifier training data  to `horizon_scout/make_training_data.py`
* Getters for horizon scout classifier training data to `getters/horizon_scout.py`
* S3 uploading functions to `discovery_utils/utils/s3.py`
* Ability to use `jsonl` file types with S3 downloading functions

### CODE REVIEW

File: discovery_utils/horizon_scout/make_training_data.py [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    afs_positive_training_data = load_and_process_positive_training_data(DataPaths.paths["AFS_POS"])
+    ahl_positive_training_data = load_and_process_positive_training_data(DataPaths.paths["AHL_POS"])
+    asf_positive_training_data = load_and_process_positive_training_data(DataPaths.paths["ASF_POS"])
+
+    # Load crunchbase orgs data which will used to sample negative training data
+    cb_orgs = (
+        s3._download_obj(
+            s3_client=client,
+            bucket=s3.BUCKET_NAME_RAW,
+            path_from="data/crunchbase/Crunchbase_2024-04-17/organizations.parquet",
```
Comment (beingkk) [inline discovery_utils/horizon_scout/make_training_data.py]: Can this be defined at the top as a global variable so it's more visible which version of the data we used for creating the training data?

File: discovery_utils/horizon_scout/make_training_data.py [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+        )
+        .query("short_description.notna()")[["id", "short_description"]]
+        .rename(columns={"short_description": "text"})
+    )
+
+    # Load GtR projects which will used to sample negative training data
+    gtr_projects_dict = s3._download_obj(
+        s3_client=client,
+        bucket=s3.BUCKET_NAME_RAW,
+        path_from="data/GtR/GtR_20240109/gtr_projects.json",
```
Comment (beingkk) [inline discovery_utils/horizon_scout/make_training_data.py]: Same comment: can this be defined at the top as a global variable so it's more visible which version of the data we used for creating the training data?

**Review by beingkk — COMMENTED** [review]
Hey @Jack-0-0 thanks for this - I tested that I can run it as well. Looks good to merge! (a couple minor comments below)

**Review by beingkk — APPROVED** [review]

**Comment by Jack-0-0** [comment]
> Hey @Jack-0-0 thanks for this - I tested that I can run it as well. Looks good to merge! (a couple minor comments below)

Thanks for the quick review. I have made the changes you suggested.

**Pull request merged**